bee-message/src/payload/transaction/mod.rs

```diff
@@ -115,10 +71,25 @@ impl Packable for TransactionPayload {
         let essence = TransactionEssence::unpack::<_, VERIFY>(unpacker)?;
         let unlock_blocks = UnlockBlocks::unpack::<_, VERIFY>(unpacker)?;
 
-        Self::builder()
-            .with_essence(essence)
-            .with_unlock_blocks(unlock_blocks)
-            .finish()
-            .map_err(UnpackError::Packable)
+        if VERIFY {
+            verify_essence_unlock_blocks(&essence, &unlock_blocks).map_err(UnpackError::Packable)?;
+        }
+
+        Ok(TransactionPayload { essence, unlock_blocks })
     }
 }
+
+fn verify_essence_unlock_blocks(essence: &TransactionEssence, unlock_blocks: &UnlockBlocks) -> Result<(), Error> {
```

Comment:
This only verifies the amount of unlock_blocks, will other validation be added later?

Reply:
If something else is added to the TIP yes.
Do you have anything in mind ?

Reply:
```
Signature must contain an Ed25519 Signature.
The Signature Unlock Block must be unique, i.e. there must not be any other Signature Unlock Blocks in the Unlock Blocks field of the transaction payload with the same signature.
```
```
Unlock syntactic validation
The Reference Unlock Block at index i must have Reference < i and the unlock block at index Reference must be a Signature Unlock Block.
```
(this should also be for alias and nft unlock blocks, right? The TIP needs to be updated to add that I think

Reply:
This is done on the unlock_block level https://github.com/iotaledger/bee/blob/shimmer-develop/bee-message/src/unlock_block/mod.rs#L109